setuptools_rust/extension.py

```diff
@@ -240,7 +240,17 @@ def _metadata(self) -> "_CargoMetadata":
             ]
             if self.cargo_manifest_args:
                 metadata_command.extend(self.cargo_manifest_args)
-            self._cargo_metadata = json.loads(subprocess.check_output(metadata_command))
+
+            env = os.environ.copy()
+            try:
+                payload = subprocess.check_output(
+                    metadata_command, env=env, encoding="latin-1"
+                )
+            except subprocess.CalledProcessError as e:
+                msg = f"cargo failed with code: {e.returncode}\n{e.output}"
```

Comment:
Should this also include stderr? Also it would be nice to say `cargo metadata` instead of just `cargo`.

Reply:
As I wrote, I copied the setup from `build.py`. Happy to add `e.stderr` here (and also in `build.py` if desired)

Reply:
Made these adaptations now.